### PR TITLE
modify layout and add homepage

### PR DESCRIPTION
**Pull request by OriolAbril**
some changes to the layout of the website:
* add description on home page about versioned docs/example gallery split and how to
  navigate them
* add underline to autolinked code within codeblocks
* move the list of tags in current notebook to the right sidebar
* change display of categories to be fixed with hightlight on current ones.
* add slides to left sidebar

### CODE REVIEW

File: examples/index.md [hunk 1]
```diff
@@ -1,14 +1,79 @@
...
-This is a WIP website.
+**Welcome to the PyMC example gallery!**
 
-This homepage is temporarily empty while we format and categorize
-all example notebooks as posts. You can see the tags and
-categories on the left sidebar and reach the already updated notebooks
-from there. All notebooks are rendered in the website though, you can access them
-via the search bar.
+The PyMC example gallery is a collection of Jupyter notebooks
+about PyMC and its usage. These notebooks range can be tutorials,
```
Comment (twiecki) [inline examples/index.md:6]:
```suggestion
about PyMC and its usage. These notebooks can be tutorials,
```

File: examples/index.md [hunk 2]
```diff
@@ -1,14 +1,79 @@
...
+case studies or in-depth explanations.
+Some notebooks were originally blogposts, others were adapted from books
```
Comment (twiecki) [inline examples/index.md:8]:
```suggestion
Some notebooks were originally blog posts, others were adapted from books
```

File: examples/index.md [hunk 3]
```diff
@@ -1,14 +1,79 @@
...
+
+**Versioned docs**
+We publish the versioned docs directly at `docs.pymc.io`. The only additions
+to the url are the language and version of the documentation.
+
+The versioned docs are synced with PyMC releases and contain a handful of
+guides about core functionality and the API documentation among other things.
+
+**Example gallery**
+We publish the example gallery as a subproject of the versioned docs: `docs.pymc.io/projects/examples`
```
Comment (twiecki) [inline examples/index.md:26]:
```suggestion
We publish the example gallery as a sub-project of the versioned docs: `docs.pymc.io/projects/examples`
```

File: examples/index.md [hunk 4]
```diff
@@ -1,14 +1,79 @@
...
+We recommend using the example gallery as an unversioned living
+resource, but we will also provide snapshots every few months for cases such as books
+that need links to a stable resource.
+
+You can access the snapshots from the version menu at the bottom right of the page.
+A version number in the `YYYY.0M.MICRO` format identifies the time the snapshot was published.
+
+---
+
+Notebooks are treated as blogposts. The metadata of each notebook
```
Comment (twiecki) [inline examples/index.md:44]:
```suggestion
Notebooks are treated as blog posts. The metadata of each notebook
```

File: examples/index.md [hunk 5]
```diff
@@ -1,14 +1,79 @@
...
+describes its topics and type of content via tags and categories
+and the last update date. We believe that tags and categories
+ease and improve navigation (as opposed to a fixed topic division/hierarchy).
+In addition, we also provide a list of recent updates and a search bar in the
+navigation bar at the top of the page.
+
+:::{caution}
+The website is still in construction. Thus, not all notebooks have been updated
```
Comment (twiecki) [inline examples/index.md:52]:
```suggestion
The website is still under construction. Thus, not all notebooks have been updated
```

File: examples/index.md [hunk 6]
```diff
@@ -1,14 +1,79 @@
...
+to include all the relevant metadata. Those notebooks can only be reached
+from the search bar
```
Comment (twiecki) [inline examples/index.md:54]:
```suggestion
from the search bar.
```

**Review by Sayam753 — APPROVED** [review]
LGTM 🚀

**Pull request merged**